src/dawn/Optimizer/PassIntervalPartitioner.cpp

```diff
@@ -0,0 +1,71 @@
+﻿//===--------------------------------------------------------------------------------*- C++ -*-===//
```

Comment:
You will need to explain me the idea of this pass.

Reply:
Assume we have a multistage with intervals [start : end], [start+1 : end-1] [start : start+4] [end-2 : end]

The pass will split those into disjoint intervals:

[start : start], [start+1 : start+4], [start+5 : end-3], [end-2 : end-1], [end : end] and will add multiple do-methods with these intervals for each computation